### PR TITLE
Update contract naming and generate new typescript types.

### LINKED ISSUE #218: Export new types for v1 contracts.
Some schemas as out of date so this will involve updating those schemas and then generating the types.

### PR DESCRIPTION
**Pull request by ezekiiel**
Resolves #218 .

This updates the contract naming to match the [updated design spec](https://github.com/DA0-DA0/dao-contracts/wiki/DAO-DAO-v1-Contracts-Design). It also updates the `types` package to include the DAO DAO v1 contract types so we can start building frontend for this.

### CODE REVIEW

**Comment by ezekiiel** [comment]
Will rebase and merge this once the named groups work gets landed. Don't want to pull things out under the feet of that work.

**Review by JakeHartnell — APPROVED** [review]

**Pull request merged**
